offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/OfferToPDFConverter.groovy

```diff
@@ -177,18 +170,41 @@ class OfferToPDFConverter implements OfferExporter {
         //and remove the footer on the first page
         //htmlContent.getElementById("grid-table-footer").remove()
 
-        List<ProductItem> listOverheadItems = offer.itemsWithOverhead
-        List<ProductItem> listNoOverheadItems = offer.itemsWithoutOverhead
+        List<ProductItem> productItems = offer.items
+
+        List<ProductItem> dataGenerationItems = []
+        List<ProductItem> dataAnalysisItems = []
+        //Project Management and Data Storage are grouped in the same category in the final Offer PDF
+        List<ProductItem> dataManagementItems = []
+
+        // Sort ProductItems into "DataGeneration", "Data Analysis" and "Project & Data Management"
+        productItems.each {
+            if (it.product instanceof Sequencing) {
+                dataGenerationItems.add(it)
+            }
+            if (it.product instanceof PrimaryAnalysis || it.product instanceof SecondaryAnalysis) {
+                dataAnalysisItems.add(it)
+            }
+            if (it.product instanceof DataStorage || it.product instanceof ProjectManagement){
+                dataManagementItems.add(it)
+            }
+        }
+
+        //Sort Lists by ProductName
+        dataGenerationItems = dataGenerationItems.sort{it.product.productName}
+        dataAnalysisItems = dataAnalysisItems.sort{it.product.productName}
+        dataManagementItems = dataManagementItems.sort{it.product.productName}
 
         //Initialize Number of table
         tableCount = 1
 
         //Initialize Count of ProductItems in table
         tableItemsCount = 1
         int maxTableItems = 8
-        //Generate ProductTable for Overhead and Non-Overhead Product Items
-        generateProductTable(OVERHEAD, listOverheadItems, maxTableItems)
-        generateProductTable(NO_OVERHEAD, listNoOverheadItems, maxTableItems)
+        //Generate ProductTable for "DataGeneration", "Data Analysis" and "Project & Data Management"  Product Items
+        generateProductTable("Data Generation", dataGenerationItems, maxTableItems)
+        generateProductTable("Data Analysis", dataAnalysisItems, maxTableItems)
+        generateProductTable("Project Management and Data Storage", dataManagementItems, maxTableItems)
```

Comment:
can then be: `generateProductTables(itemsByGroup)`

Reply:
Done!